diff --git a/src/fiber/fiber.cr b/src/fiber/fiber.cr
index 41c4aa3..5cb3291 100644
--- a/src/fiber/fiber.cr
+++ b/src/fiber/fiber.cr
@@ -67,7 +67,7 @@ class Fiber
     @@stack_pool.pop? || LibC.mmap(nil, Fiber::STACK_SIZE,
       LibC::PROT_READ | LibC::PROT_WRITE,
       LibC::MAP_PRIVATE | LibC::MAP_ANON,
-      -1, LibC::SSizeT.new(0)).tap do |pointer|
+      -1, 0).tap do |pointer|
       raise Errno.new("Cannot allocate new fiber stack") if pointer == LibC::MAP_FAILED
       ifdef linux
         LibC.madvise(pointer, Fiber::STACK_SIZE, LibC::MADV_NOHUGEPAGE)
diff --git a/src/libc.cr b/src/libc.cr
index 964c5ad..39866dc 100644
--- a/src/libc.cr
+++ b/src/libc.cr
@@ -23,7 +23,12 @@ lib LibC
   alias PtrDiffT = SSizeT
   alias TimeT = SSizeT
   alias PidT = Int
-  alias OffT = SSizeT
+
+  ifdef musl
+    alias OffT = Int64
+  else
+    alias OffT = SSizeT
+  end
 
   ifdef darwin
     alias ModeT = UInt16
@@ -67,7 +72,7 @@ lib LibC
 
   MAP_FAILED = Pointer(Void).new(SizeT.new(-1))
 
-  fun mmap(addr : Void*, len : SizeT, prot : Int, flags : Int, fd : Int, offset : SSizeT) : Void*
+  fun mmap(addr : Void*, len : SizeT, prot : Int, flags : Int, fd : Int, offset : OffT) : Void*
   fun munmap(addr : Void*, len : SizeT)
   fun madvise(addr : Void*, len : SizeT, advise : Int) : Int
   fun mprotect(addr : Void*, len : SizeT, prot : Int) : Int
